### PR TITLE
clarify named parameter discussion

### LINKED ISSUE #1714: 'A tour of the Dart language' page issue
Page URL: https://www.dartlang.org/guides/language/language-tour
Page source: https://github.com/dart-lang/site-www/tree/master/src/_guides/language/language-tour.md

Found a typo? You can fix it yourself by going to the page source and clicking the pencil icon. Or finish creating this issue.

Description of issue:

I somewhat realized that named parameters in dart are optional by default, unless annotated with '@required'.

This is far from clear by reading this documentation, especially the following line:

"A function can have two types of parameters: required and optional. The required parameters are listed first, followed by any optional parameters. Named optional parameters can also be marked as @required. See the next section for details."

### PR DESCRIPTION
**Pull request by kwalrath**
Fixes #1714.

It's rather difficult to talk about this, but I tried to make it clearer.

I also did a little bit of cleanup elsewhere, while I was in the neighborhood.

### CODE REVIEW

File: src/_guides/language/language-tour.md [hunk 1]
```diff
@@ -1136,25 +1136,31 @@ The <code>=> <em>expr</em></code> syntax is a shorthand for
...
-statement](#if-and-else) there, but you can use a [conditional
-expression](#conditional-expressions).
-</div>
+<aside class="alert alert-info" markdown="1">
+  **Note:**
+  Only an *expression*—not a *statement*—can appear between the arrow
+  (=\>) and the semicolon (;). For example, you can’t put an [if
+  statement](#if-and-else) there, but you can use a [conditional
+  expression](#conditional-expressions).
+</aside>
```
Comment (sfshaza2) [inline src/_guides/language/language-tour.md]: Does dart.dev support {{site.alert.info}} / {{site.alert.end}}  ? That’s what we’re using now on flutter.dev. And, yes, the content is indented by 2 chars.  :)  Also, the macros add the **Note:** or **Warning** text, as the case may be.  (Depending on whether it’s .info or .warning, for example.)

File: src/_guides/language/language-tour.md [hunk 2]
```diff
@@ -1136,25 +1136,31 @@ The <code>=> <em>expr</em></code> syntax is a shorthand for
...
-See the next section for details.
+A function can have two types of parameters: _required_ and _optional_.
+The required parameters are listed first, followed by any optional parameters.
+Optional parameters can be _named_ or _positional_.
 
+<aside class="alert alert-info" markdown="1">
+  **Note:**
+  Some APIs — notably [Flutter][] widget constructors —
+  use only optional named parameters,
+  even for parameters that are mandatory.
```
Comment (sfshaza2) [inline src/_guides/language/language-tour.md]: use only optional named parameters, even for params that are mandatory ?   This sentence confuses me…

Comment (kwalrath) [reply src/_guides/language/language-tour.md]: That's because it's inherently confusing. :/

File: src/_guides/language/language-tour.md (hunk 1 above)
Comment (kwalrath) [reply src/_guides/language/language-tour.md]: Nope. I'll file a bug about that.

File: src/_guides/language/language-tour.md (hunk 2 above)
Comment (kwalrath) [reply src/_guides/language/language-tour.md]: I've changed "optional named" to "named" here.

**Pull request merged**